src/components/ec-btn-dropdown/ec-btn-dropdown.vue

```diff
@@ -44,55 +44,57 @@
   </div>
 </template>
 
-<script>
+<script setup>
+import { ref } from 'vue';
+
 import EcBtn from '../ec-btn';
 import EcDropdownSearch from '../ec-dropdown-search';
 
-export default {
-  name: 'EcBtnDropdown',
-  components: {
-    EcBtn,
-    EcDropdownSearch,
+const props = defineProps({
+  items: {
+    type: Array,
+    default: () => [],
+  },
+  isDisabled: {
+    type: Boolean,
+    default: false,
   },
-  props: {
-    items: {
-      type: Array,
-      default: () => [],
-    },
-    isDisabled: {
-      type: Boolean,
-      default: false,
-    },
-    buttonText: {
-      type: String,
-      required: true,
-    },
-    listDataTest: {
-      type: String,
-    },
+  buttonText: {
+    type: String,
+    required: true,
   },
-  emits: ['click', 'change'],
-  data() {
+  listDataTest: {
+    type: String,
+  },
+});
+
+const emit = defineEmits(['click', 'change']);
+
+const isOpen = ref(false);
+const popoverOptions = ref({
```

Comment:
I noticed that this object is static and can't change in the future. We can optimize that either by removing the `ref()` so the reactivity doesn't have to keep track of it, or we can just inline the object into the template `:popover-options="{ autoSize: 'min', placement: 'bottom-end', }"`

Reply:
corrected